### PR TITLE
implementation of PAL for Zephyr RTOS based on efm32pg stk3402a board

### PR DESCRIPTION
**Pull request by panmasuo**
Hi,
My current project in Arrow Electronics is implementing demo based on Optiga Trust X and [Zephyr RTOS](https://docs.zephyrproject.org/latest/introduction/index.html).

For that i created PAL Layer for board efm32pg-stk3402a based on that OS.

Demo is working correctly (using examples shown in /example folder) so i would like to share my progress on this repo.

I would like to get feedback about that Pull Request and i can answer more questions about project/demo.
Thanks
Pawel Zarembski

### CODE REVIEW

**Comment by ayushev** [comment]
Thank you @panmasuo for your commitment!
The example indeed looks very good. Please have a look on my comments, and let me know.
We would like to publish as well a link to the explanation/howto guide of the setup (EFM32 + Zephy OS + Trust X), do you have it? We could then give a reference to it under "Partner's Application Notes"

**Comment by ayushev** [comment]
@panmasuo a question. Do you have a recommendation on which part of the [Porting Guide](https://github.com/Infineon/optiga-trust-x/wiki/Porting-Guide) could be improved. Where did you have most of complications during the porting.

File: pal/efm32pg_zephyr/pal_i2c.c [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+            (app_event_handler_t)p_i2c_context->upper_layer_event_handler;
+
+    if (pal_i2c_acquire((void*)p_i2c_context) == PAL_STATUS_SUCCESS) {
+        int result;
+
+        result = i2c_write(i2c_dev, p_data, (u32_t)length,
+                           p_i2c_context->slave_address);
+
+        if (result == 0) {
+            upper_layer_handler(p_i2c_context->upper_layer_ctx,
```
[the remark this answers is not in the document]
Comment (panmasuo) [reply pal/efm32pg_zephyr/pal_i2c.c]: Yes, that is correct. But i don't understand how does this affect the code? 
i2c_write is returning 0 if successful and other value if error andthose two cases are covered
[I2C - Zephyr](https://docs.zephyrproject.org/latest/reference/peripherals/i2c.html)

**Comment by panmasuo** [comment]
> We would like to publish as well a link to the explanation/howto guide of the setup (EFM32 + Zephy OS + Trust X), do you have it? We could then give a reference to it under "Partner's Application Notes"

I don't have such a guide. I can create something like this but first i would like to begin conversation with Zephyr Project Team about including Optiga Trust X library to their OS as either library, external library or module. 
For users it would be much easier because right now to include Optiga library you need to create Kconfig/Cmake files for compile it correctly.

**Comment by panmasuo** [comment]
> Do you have a recommendation on which part of the [Porting Guide](https://github.com/Infineon/optiga-trust-x/wiki/Porting-Guide) could be improved. Where did you have most of complications during the porting.
@ayushev 
I don't remember all the details right now because i started working with Optiga few moths ago, but i will try to list things that gave me some trouble.

- Pins used in Optiga - basically i had a lot of problem with recognizing which pins of Optiga are used, and where are they (i am using Optiga Trust X shield from XMC4500 Relax Kit). There is no documentation about that Shield anywhere so i had to look which pins are used in DAVE, then look for them on XMC, then transfer that to shield. Some documentation about that would be awesome.
- Work involved with adding mbedTLS was troublesome because (if i remember correctly) custom config file/header was somehow hidden in DAVE generators (and i was basing on that project). Maybe adding some guide about integrating mbedTLS to Optiga would be helpful. 
- Sometimes it wasn't clear which of the functions/variables/macros are from Optiga library and which from platform (in PAL implementations).  Adding some "empty PAL implementation" would help at beginning to see what is needed and what is extra.

If something else will pop into my mind i will get you know.

File: pal/efm32pg_zephyr/pal_i2c.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (ayushev) [reply pal/efm32pg_zephyr/pal_i2c.c]: I see, the documentation says the i2c_write function is synchronous, so it's fine.

**Pull request merged**